### PR TITLE
Add user create with option of rotation

### PR DESCRIPTION
**Pull request by giarc3**
- Added `userCreate` functions to `IronSdk`, `IronSdkSync`, and `IronSdkFuture`. It takes a jwt, password, and options
- Added `UserCreateOpts`, which contains the desired `needsRotation`
- Added `UserCreateResult`
- Added `PublicKey`, since it is a part of `UserCreateKeyPair`
- Wrote a test for an invalid jwt
- Can't write a passing test because of jwt reasons (scary)

Note: I'm confused by our scaladocs, as I can't seem to view the @param or @return for any of our methods when I view the docs.

### CODE REVIEW

File: src/main/scala/com/ironcorelabs/UserCreateKeyPair.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.ironcorelabs.scala.sdk
+
+import com.ironcorelabs.{sdk => jsdk}
+
+case class UserCreateKeyPair(
```
Comment (coltfred) [inline src/main/scala/com/ironcorelabs/UserCreateKeyPair.scala]: I think we should change this name. UserCreateResult seems best. We should change it all the way down to ironoxide IMO.

File: src/main/scala/com/ironcorelabs/UserCreateKeyPair.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  userEncryptedMasterKey: PrivateKey,
+  userPublicKey: PublicKey,
+  needsRotation: Boolean
+)
+
+object UserCreateKeyPair {
+  def apply(uckp: jsdk.UserCreateKeyPair): UserCreateKeyPair =
+    UserCreateKeyPair(
+      PrivateKey(uckp.getUserEncryptedMasterKey),
```
Comment (coltfred) [inline src/main/scala/com/ironcorelabs/UserCreateKeyPair.scala]: This isn't a private key. It's an encrypted private key. We should use a type that describes that.

File: src/main/scala/com/ironcorelabs/UserCreateOpts.scala [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.ironcorelabs.scala.sdk
+
+import cats.effect.Sync
+import com.ironcorelabs.{sdk => jsdk}
+
+case class UserCreateOpts(
```
Comment (coltfred) [inline src/main/scala/com/ironcorelabs/UserCreateOpts.scala]: nitpick: this case class can be one line.

File: src/test/scala/ironoxide/FullIntegrationTest.scala [hunk 4]
```diff
@@ -25,6 +25,8 @@ class FullIntegrationTest extends AsyncWordSpec with Matchers with EitherValues
   // Hardcoded user info for these tests because they don't depend on the number of things created for the given user, just
   // the values created.
   val primaryTestUserId = UserId("b29c1ee7-ede9-4401-855a-3a78a34a2759")
+  val primaryTestUserPassword = java.util.UUID.randomUUID().toString()
```
Comment (coltfred) [inline src/test/scala/ironoxide/FullIntegrationTest.scala]: unused.

File: src/main/scala/com/ironcorelabs/IronSdk.scala [hunk 5]
```diff
@@ -50,10 +50,23 @@ trait IronSdk[F[_]] {
...
    *
    * @return an instance of the [[IronSdkAdvanced]]
    */
   def advanced: IronSdkAdvanced[F]
 }
+
+object IronSdk {
+  import cats.effect.Sync
+  import cats.implicits._
+  def userCreate[F[_]](jwt: String, password: String, options: UserCreateOpts)(
```
Comment (coltfred) [inline src/main/scala/com/ironcorelabs/IronSdk.scala]: Would you add a doc comment to this, please.

File: src/main/scala/com/ironcorelabs/IronSdk.scala [hunk 6]
```diff
@@ -50,10 +50,23 @@ trait IronSdk[F[_]] {
    */
   def groupCreate(options: GroupCreateOpts): F[GroupMetaResult]
 
+  def userCreate(jwt: String, password: String, options: UserCreateOpts): F[UserCreateKeyPair]
```
Comment (coltfred) [inline src/main/scala/com/ironcorelabs/IronSdk.scala]: Comment, please.

Comment (giarc3) [reply src/main/scala/com/ironcorelabs/IronSdk.scala]: Added

File: src/main/scala/com/ironcorelabs/IronSdk.scala (hunk 5 above)
Comment (giarc3) [reply src/main/scala/com/ironcorelabs/IronSdk.scala]: Added

File: src/main/scala/com/ironcorelabs/UserCreateKeyPair.scala (hunk 1 above)
Comment (giarc3) [reply src/main/scala/com/ironcorelabs/UserCreateKeyPair.scala]: Changed to `UserCreateResult`

File: src/main/scala/com/ironcorelabs/UserCreateKeyPair.scala (hunk 2 above)
Comment (giarc3) [reply src/main/scala/com/ironcorelabs/UserCreateKeyPair.scala]: After talking more, this `userEncryptedMasterKey` was removed from `UserCreateResult`

File: src/main/scala/com/ironcorelabs/UserCreateOpts.scala (hunk 3 above)
Comment (giarc3) [reply src/main/scala/com/ironcorelabs/UserCreateOpts.scala]: Fixed

File: src/test/scala/ironoxide/FullIntegrationTest.scala (hunk 4 above)
Comment (giarc3) [reply src/test/scala/ironoxide/FullIntegrationTest.scala]: Removed

**Pull request merged**